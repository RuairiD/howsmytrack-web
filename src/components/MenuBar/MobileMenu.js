import React from "react";

import { Collapse, Icon } from "antd";

import MobileHeader from "./MobileHeader";
import MainMenu from "./MainMenu";

const MobileMenu = ({
    onMenuClick,
    username,
    rating,
    notifications,
    onCollapseChange,
    mobileMenuCollapsed,
}) => (
    <Collapse
        className="mobile-menu-collapse"
        bordered={false}
        expandIcon={({ isActive }) => (
            <Icon type="menu" rotate={isActive ? 90 : 0} />
        )}
        onChange={onCollapseChange}
    >
        <Collapse.Panel
            header={(
                <MobileHeader
                    mobileMenuCollapsed={mobileMenuCollapsed}
                    notifications={notifications}
                />
            )}
            key="1"
        >
            <MainMenu
                isMobile
                onMenuClick={onMenuClick}
                username={username}
                rating={rating}
                notifications={notifications}
            />
        </Collapse.Panel>
    </Collapse>
);

export default MobileMenu;
