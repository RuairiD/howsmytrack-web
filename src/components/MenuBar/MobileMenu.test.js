import React from "react";
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import MobileMenu from "./MobileMenu";

describe("MobileMenu", () => {
    it("renders a MainMenu within a Collapse to allow mobile users to show/hide the menu at will", () => {
        const wrapper = shallow(
            <MobileMenu
                onMenuClick={() => {}}
                username="username"
                rating={4.321}
                notifications={2}
                onCollapseChange={() => {}}
                mobileMenuCollapsed
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders an unrotated menu icon in the Collapse header when collapsed", () => {
        const wrapper = mount(
            <MobileMenu
                onMenuClick={() => {}}
                username="username"
                rating={4.321}
                notifications={2}
                onCollapseChange={() => {}}
                mobileMenuCollapsed
            />,
        );

        expect(wrapper.find("Collapse").get(0).props.expandIcon({ isActive: false }).props.rotate).toBe(0);
    });

    it("renders a rotated menu icon in the Collapse header when uncollapsed", () => {
        const wrapper = mount(
            <MobileMenu
                onMenuClick={() => {}}
                username="username"
                rating={4.321}
                notifications={2}
                onCollapseChange={() => {}}
                mobileMenuCollapsed
            />,
        );

        expect(wrapper.find("Collapse").get(0).props.expandIcon({ isActive: true }).props.rotate).toBe(90);
    });
});
