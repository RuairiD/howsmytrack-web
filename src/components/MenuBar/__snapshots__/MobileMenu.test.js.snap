// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileMenu renders a MainMenu within a Collapse to allow mobile users to show/hide the menu at will 1`] = `
<Collapse
  bordered={false}
  className="mobile-menu-collapse"
  expandIcon={[Function]}
  expandIconPosition="left"
  onChange={[Function]}
  openAnimation={
    Object {
      "appear": [Function],
      "enter": [Function],
      "leave": [Function],
    }
  }
>
  <CollapsePanel
    header={
      <MobileHeader
        mobileMenuCollapsed={true}
        notifications={2}
      />
    }
    key="1"
  >
    <MainMenu
      isMobile={true}
      notifications={2}
      onMenuClick={[Function]}
      rating={4.321}
      username="username"
    />
  </CollapsePanel>
</Collapse>
`;
